src/main.py

```diff
@@ -76,7 +78,11 @@ def execute_block(block_name):
     if is_resuming:
         result = block.resume(body['state'], cleaned_params)
```

Comment:
Should catch exceptions from this as well, so just wrap the try around the entire if-else structure